### ISSUE TITLE
Does not compile with ghc-9.6

### ISSUE THREAD

**Issue by erikd** [kowainik/validation-selective#76]
This package depends on `selective < 0.7` and that does not compile with `ghc-9.6`.

I am currently getting around this in my `cabal.project` file using:
```
if impl(ghc >= 9.6)
  allow-newer:
    , validation-selective:selective
```
to allow the use of `selective == 0.7`.

**Comment by tomjaguarpaw** [comment]
`select-0.6` now seems to compile with 9.6. Perhaps it had a restrictive upper-bound which has now been relaxed with a Hackage revision?

**Comment by erikd** [comment]
Yes, the Hackage version now works, but this issue should remain open until it has also been fixed in this repo.

**Comment by andreasabel** [comment]
Here is a revision: https://hackage.haskell.org/package/validation-selective-0.2.0.0/revisions/
- tested with GHC 9.6.2
- built with GHC 9.8.1-alpha1

Maybe a bit generous for the nervous packages `base`, `hspec` and `doctest`, but since this is a now unmaintained package, let's take the lazy approach.

**Comment by tomjaguarpaw** [comment]
> Yes, the Hackage version now works, but this issue should remain open until it has also been fixed in this repo.

Well, the issue title is now incorrect. It does compile with 9.6. Perhaps, to avoid confusion, there should be a new ticket to bump the dependency of `selective` to `< 0.7` (if that is a valid thing to do, I haven't checked).
